### DISCUSSION TITLE
Significant warnings when using bs4Dash

### DISCUSSION THREAD

**Discussion by florianm** [ThinkR-open/golem#849] [Q&A]
Hi, 

I've started a Shiny app using bs4Dash from the Golem template. 
Golem is a wonderful template and saves me a lot of blood, sweat, and tears getting the initial app up and running. bs4Dash provides a great front-end toolkit but overwrites many shiny functions.

The following "significant warnings" break my R CMD Check:

```
Warning: Found the following significant warnings:
  Warning: replacing previous import ‘bs4Dash::insertTab’ by ‘shiny::insertTab’ when loading ‘turtleviewer2’
  Warning: replacing previous import ‘bs4Dash::actionButton’ by ‘shiny::actionButton’ when loading ‘turtleviewer2’
  Warning: replacing previous import ‘bs4Dash::tabsetPanel’ by ‘shiny::tabsetPanel’ when loading ‘turtleviewer2’
  Warning: replacing previous import ‘bs4Dash::column’ by ‘shiny::column’ when loading ‘turtleviewer2’
```
GHA logs e.g. [here](https://github.com/dbca-wa/turtleviewer2/runs/5458859081?check_suite_focus=true)

Has anyone seen warnings like these? What could I check next to find the cause?
Is there a fix apart from namespacing every single shiny function I'm using and dropping the blanket `@import shiny`?

**Update**
I have resolved the warnings by replacing the blanket `@import shiny` statements with `@importFrom shiny ...`. 
The blanket imports make sense for a "vanilla" Golem app using pure `shiny` functions, and the collisions are more a problem on the `bs4Dash` side.
I hope leaving this solution here might help others.

## Details

<details>

* Code repo https://github.com/dbca-wa/turtleviewer2/
* The four colliding functions are nowhere used directly in the entire package (search comes up blank)
* The four colliding functions are not explicitly mentioned in my [NAMESPACE](https://github.com/dbca-wa/turtleviewer2/blob/main/NAMESPACE)
* There are blanket `@import shiny` and `@import bs4Dash` statements across the entire app
* I'm not using modules
* I'm not using golem ui helpers
* web searches come up blank
</details>

**Comment by amygimma** [comment]
I'm hitting this same problem- i'd rather not need to import each function from shiny individually - is there another way to resolve these warnings?
